### ISSUE TITLE
moveTo on iconItem causes garbage in preview

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#3951]
-> @peller but any of us could look at it, too

@billreed63 @jhpedemonte 

Same problem with M9, so I figure we can defer this if we don't find a quick and safe fix.
- Open Sample3Mobile.html
- Save into root folder as a1.html (or whatever)
- Add an IconContainer
- Select first IconItem
- Set moveTo to `alerts_view`
- Preview
- Click on the first IconItem

Result=>

![Screen Shot 2013-04-24 at 10 06 39 AM](https://f.cloud.github.com/assets/706721/421163/bab6ce1e-ad01-11e2-8f10-f2c6373056ac.png)

**Comment by peller** [comment]
I suppose we strip out as much code as we can (e.g. our states code references) and submit to Dojo.  First let's see if the problem still exists when run against Dojo 1.9

**Comment by jhpedemonte** [comment]
Well, there's something that is just odd about this whole testcase:  there are 2 top-level sibling Views.  Then in one of the Views we add an IconContainer and tell it to switch to the other View when clicking on one of the IconItems.

I think this is outside the bounds of how the widgets were intended to interact.  I say close this as "won't fix".

**Comment by JonFerraiolo** [comment]
Why is the `moveTo` property on an IconItem different than a `moveTo` property on a ListItem,TabBarButton, Heading or ToolBarButton? For the other 4 cases, setting `moveTo` to the name of another view will cause the application to switch to the other view. 

I discovered this when fixing #3948 (where the moveTo property in Widget palette wasn't showing mobile views for files migrated from M9). There are 5 `moveTo` properties. I verified that ListItem,TabBarButton, Heading and ToolBarButton all work correctly, but IconItem does weird things.

**Comment by JonFerraiolo** [comment]
And the documentation in file IconItem.js claims the `moveTo` property will switch to a different view.

**Comment by JonFerraiolo** [comment]
Also broken in Dojo 1.9.rc1 when there is no Maqetta stuff in there.

I'll file a dojo ticket.

**Comment by JonFerraiolo** [comment]
Dojo ticket http://bugs.dojotoolkit.org/ticket/17073

**Comment by JonFerraiolo** [comment]
Closing - bascially it is working "as designed".

I debugged into Dojo and figured out that at least part of the problem was that the `transition` property was set to `below`. If I entered `slide` as the new value, then the views did the appropriate slide-in/slide-out.

The Dojo docs say that the default is 'slide', but that's not entirely correct. The 'transition' property (and several others) get their default values from their parent widget (via property `paramsToInherit` in IconItem.js), and the parent widget (IconContainer) sets the default value to `below`. Maqetta pulls the property value by doing a `get()` on the dojo widget, and the widget tells us the current value is `below`.

I'm not sure exactly what to do in terms of helping users. Probably nothing since Maqetta is honoring Dojo's defaults. I suspect that when users run into this, someone who has run into this situation previously (e.g., me) might tell the user about the problem via the user forums.
